### ISSUE TITLE
Error [ERR_REQUIRE_ESM]: require() of ES Module ... not supported

### ISSUE THREAD

**Issue by goldylucks** [nrwl/nx#12634]
## Current Behavior

From an express app, importing some packages throw this error.

## Expected Behavior

Should just work

## Steps to Reproduce

```js
// main.ts
import "react-dnd"
```

```
$ npx create-nx-workspace foo
$ cd foo
$ yarn add react-dnd
$ yarn serve
```

### Failure Logs
```
/Users/Goldy/breathe/code/node_modules/@nrwl/node/src/executors/node/node-with-require-overrides.js:16
        return originalLoader.apply(this, arguments);
                              ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/Goldy/breathe/code/node_modules/react-dnd/dist/index.js from /Users/Goldy/breathe/code/dist/apps/admin-ui/main.js not supported.
Instead change the require of index.js in /Users/Goldy/breathe/code/dist/apps/admin-ui/main.js to a dynamic import() which is available in all CommonJS modules.
```

### Environment

```
   Node : 17.6.0
   OS   : darwin arm64
   yarn : 1.22.18

   nx : 14.4.3
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.4.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.4.3
   @nrwl/eslint-plugin-nx : 14.4.3
   @nrwl/express : 14.4.3
   @nrwl/jest : 14.4.3
   @nrwl/js : 14.4.3
   @nrwl/linter : 14.4.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : 14.4.3
   @nrwl/nx-cloud : 14.2.0
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.4.3
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.4.3
   @nrwl/web : 14.4.3
   @nrwl/workspace : 14.4.3
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:
```

**Comment by mikehaas763** [comment]
@goldylucks I'm guessing your root `package.json` has `"type": "module"` in it?

**Comment by AgentEnder** [comment]
This will be partially fixed by #10414, and further addressed in the future.
